### PR TITLE
KOGITO-6490: kogito-addons-quarkus-tracing-decision: LocalDate* are on available in the ObjectMapper

### PR DESCRIPTION
**Pull request by manstis**
See https://issues.redhat.com/browse/KOGITO-6490

This is an attempt at a more generalised solution following #1762 

This will need refactoring after https://github.com/kiegroup/kogito-runtimes/pull/1828 is merged.

Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](CONTRIBUTING.md)
- [x] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config)
- [x] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [x] Pull Request title contains the target branch if not targeting main: `[0.9.x] KOGITO-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>Run all builds</b>  
  Please add comment: <b>Jenkins retest this</b>

* <b>Run (or rerun) specific test(s)</b>  
  Please add comment: <b>Jenkins (re)run [kogito-runtimes|optaplanner|kogito-apps|kogito-examples|optaplanner-quickstarts|optaweb-employee-rostering|optaweb-vehicle-routing] tests</b>
 
* <b>Quarkus LTS checks</b>  
  Please add comment: <b>Jenkins run LTS</b>

* <b>Run (or rerun) LTS specific test(s)</b>  
  Please add comment: <b>Jenkins (re)run [kogito-runtimes|optaplanner|kogito-apps|kogito-examples|optaplanner-quickstarts|optaweb-employee-rostering|optaweb-vehicle-routing] LTS</b>

* <b>Native checks</b>  
  Please add comment: <b>Jenkins run native</b>

* <b>Run (or rerun) native specific test(s)</b>  
  Please add comment: <b>Jenkins (re)run [kogito-runtimes|optaplanner|kogito-apps|kogito-examples|optaplanner-quickstarts|optaweb-employee-rostering|optaweb-vehicle-routing] native</b>

* <b>Full Kogito testing</b> (with cloud images and operator BDD testing)  
  Please add comment: <b>Jenkins run BDD</b>  
  <b>This check should be used only if a big change is done as it takes time to run, need resources and one full BDD tests check can be done at a time ...</b>
</details>

### CODE REVIEW

File: kogito-codegen-modules/kogito-codegen-core/pom.xml [hunk 1]
```diff
@@ -38,6 +38,12 @@
             <artifactId>jackson-databind</artifactId>
         </dependency>
 
+        <!-- required because of JSR310 support for LocalDate -->
+        <dependency>
+            <groupId>com.fasterxml.jackson.datatype</groupId>
+            <artifactId>jackson-datatype-jsr310</artifactId>
+        </dependency>
```
Comment (danielezonca) [inline kogito-codegen-modules/kogito-codegen-core/pom.xml:45]: Codegen is a compile type module so I don't think it is enough to add this dependency here.
Can you add an integration test with DMN to see if works end-to-end?

Comment (manstis) [reply kogito-codegen-modules/kogito-codegen-core/pom.xml]: @danielezonca I have moved the declaration to elsewhere (-runtime related) in a new commit.

I'm just running a build locally before looking into adding an integration test.

File: quarkus/addons/common/deployment/pom.xml [hunk 2]
```diff
@@ -18,6 +18,11 @@
       <groupId>io.quarkus</groupId>
       <artifactId>quarkus-core-deployment</artifactId>
     </dependency>
+    <dependency>
```
Comment (manstis) [inline quarkus/addons/common/deployment/pom.xml:21]: ... and I think this still may not be a runtime module!!

**Comment by manstis** [comment]
@evacchi Should we also be concerned about the implementation of `org.kie.kogito.incubation.common.objectmapper.quarkus.QuarkusInternalObjectMapper` that does not register JSR-310 support?

**Review by danielezonca — APPROVED** [review]
LGTM 👍 
Minor "styling" comments

**Review by ricardozanini — APPROVED** [review]

File: springboot/addons/tracing-decision/src/test/java/org/kie/kogito/tracing/decision/SpringBootDecisionTracingTest.java [hunk 3]
```diff
@@ -15,54 +15,12 @@
...
-    private static final ObjectMapper MAPPER = new ObjectMapper()
-            .registerModule(JsonFormat.getCloudEventJacksonModule());
-
-    private static final String TEST_MODEL_RESOURCE = "/Traffic Violation.dmn";
-    private static final String TEST_MODEL_NAMESPACE = "https://github.com/kiegroup/drools/kie-dmn/_A4BCA8B8-CF08-433F-93B2-A2598F19ECFF";
-    private static final String TEST_MODEL_NAME = "Traffic Violation";
-
-    private static final String TEST_EXECUTION_ID = "7c50581e-6e5b-407b-91d6-2ffb1d47ebc0";
-    private static final Map<String, Object> TEST_CONTEXT_VARIABLES = new HashMap<String, Object>() {
+    private static final Map<String, Object> TEST_CONTEXT_VARIABLES = new HashMap<>() {
```
[the remark this answers is not in the document]
Comment (manstis) [reply springboot/addons/tracing-decision/src/test/java/org/kie/kogito/tracing/decision/SpringBootDecisionTracingTest.java]: Done.

File: springboot/addons/tracing-decision/src/test/java/org/kie/kogito/tracing/decision/SpringBootDecisionTracingOneOfEachTypeTest.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.Period;
+import java.util.HashMap;
+import java.util.Map;
+
+public class SpringBootDecisionTracingOneOfEachTypeTest extends BaseSpringBootDecisionTracingTest {
+
+    private static final Map<String, Object> TEST_CONTEXT_VARIABLES = new HashMap<>() {
```
[the remark this answers is not in the document]
Comment (manstis) [reply springboot/addons/tracing-decision/src/test/java/org/kie/kogito/tracing/decision/SpringBootDecisionTracingOneOfEachTypeTest.java]: Done.

File: quarkus/addons/tracing-decision/runtime/src/test/java/org/kie/kogito/tracing/decision/QuarkusDecisionTracingTest.java [hunk 5]
```diff
@@ -15,47 +15,13 @@
...
-import static org.mockito.Mockito.verify;
-import static org.mockito.Mockito.when;
-
-public class QuarkusDecisionTracingTest {
-
-    public static final String TEST_MODEL_NAMESPACE = "https://github.com/kiegroup/drools/kie-dmn/_A4BCA8B8-CF08-433F-93B2-A2598F19ECFF";
     public static final String TEST_MODEL_NAME = "Traffic Violation";
+    public static final String TEST_MODEL_NAMESPACE = "https://github.com/kiegroup/drools/kie-dmn/_A4BCA8B8-CF08-433F-93B2-A2598F19ECFF";
 
     private static final Map<String, Object> TEST_CONTEXT_VARIABLES = new HashMap<String, Object>() {
```
[the remark this answers is not in the document]
Comment (manstis) [reply quarkus/addons/tracing-decision/runtime/src/test/java/org/kie/kogito/tracing/decision/QuarkusDecisionTracingTest.java]: Done.

File: quarkus/addons/tracing-decision/runtime/src/test/java/org/kie/kogito/tracing/decision/QuarkusDecisionTracingOneOfEachTypeTest.java [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.Period;
+import java.util.HashMap;
+import java.util.Map;
+
+public class QuarkusDecisionTracingOneOfEachTypeTest extends BaseQuarkusDecisionTracingTest {
+
+    private static final Map<String, Object> TEST_CONTEXT_VARIABLES = new HashMap<>() {
```
[the remark this answers is not in the document]
Comment (manstis) [reply quarkus/addons/tracing-decision/runtime/src/test/java/org/kie/kogito/tracing/decision/QuarkusDecisionTracingOneOfEachTypeTest.java]: Done.

**Review by r00ta — APPROVED** [review]
Nice work!

**Comment by manstis** [comment]
> @evacchi Should we also be concerned about the implementation of `org.kie.kogito.incubation.common.objectmapper.quarkus.QuarkusInternalObjectMapper` that does not register JSR-310 support?

This was discussed on Zulip and captured here: https://issues.redhat.com/browse/KOGITO-6512

**Comment by manstis** [comment]
Jenkins run kogito-examples tests

**Comment by manstis** [comment]
Jenkins run kogito-examples tests

**Comment by manstis** [comment]
Jenkins run kogito-examples tests

**Pull request merged**